samza-rest/src/main/java/org/apache/samza/monitor/JobsClient.java

```diff
@@ -90,41 +91,45 @@ public JobStatus getJobStatus(JobInstance jobInstance) {
    * When a job status server is down or returns a error response, it tries to reach out to
    * the next job status server in the sequence, to complete the http get request.
    *
-   * @param urlMapFunction to build the request url, given job status server base url.
+   * @param requestUrlBuilder to build the request url, given job status server base url.
    * @param <T> return type of the http get response.
    * @return the response from any one of the job status server.
    * @throws Exception when all the job status servers are unavailable.
    *
    */
-  private <T> T retriableHttpGet(Function<String, String> urlMapFunction) {
+  private <T> T retriableHttpGet(Function<String, String> requestUrlBuilder, TypeReference<T> typeReference) {
```

Comment:
Not retriable either. I'd actually prefer moving the logic for iterating over the jobStatusServers and error handling to the caller method.